### ISSUE TITLE
custom-firewall.sh: jump to firewall is appended rather than prepended

### ISSUE THREAD

**Issue by jannispl** [deis/deis#4686]
Currently, the custom firewall script appends a jump to `Firewall-INPUT` to the `INPUT` and `FORWARD` chains. This can lead to the following `FORWARD` chain:
```
Chain FORWARD (policy DROP 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination         
24310   31M DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0           
15664   16M ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
22853 2429K ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 Firewall-INPUT  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  docker0 docker0  0.0.0.0/0            0.0.0.0/0   
```
As you can see, the chain may be ACCEPTed by some docker rule without going through the firewall. This results in ports bound by docker containers (and exposed on the host) to be publically accessible.

Potential fix:
```
cat custom-firewall.sh | \
    sed 's/-A INPUT -j Firewall-INPUT/-I INPUT 1 -j Firewall-INPUT/' | \
    sed 's/-A FORWARD -j Firewall-INPUT/-I FORWARD 1 -j Firewall-INPUT/' > new-firewall.sh
```

**Comment by bacongobbler** [comment]
@jannispl would you mind PR'ing a fix for this? :)

**Comment by jannispl** [comment]
closing in favor of #4687 and #4724

**Comment by jannispl** [comment]
reopening - ping @aledbf 
After changing my cluster to start flanneld on boot, docker still uses iptables.
The command line switch responsible for forcing docker to skip iptables setup is `--iptables false`, however I'm not sure how docker will route traffic in this case.

**Comment by aledbf** [comment]
@jannispl if flannel is started with `--ip-masq=true` then you can use `--iptables false` in docker (you need to do this change in your docker.service unit). 
Check this in the file `/var/run/flannel_docker_opts.env`.

**Comment by aledbf** [comment]
@jannispl this is the next change after #4721 (flannel must be working correctly before disabling iptables)

**Comment by jannispl** [comment]
@aledbf I tried setting `--iptables=false` and realized that some containers were not able to communicate to remote hosts on the private network (e.g. container on 10.29.77.8 was not able to ping 10.29.77.9).

**Comment by bacongobbler** [comment]
This is not something we'll be able to get to for the LTS release (#4776), and provisioning Deis v2 is done through installing a vanilla kubernetes cluster so I'm going to go ahead and close this.
